### ISSUE TITLE
PyRight no longer picking up symbols in Jax library

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#5668]
This seems to be a regression in recent versions of PyRight:

```python
from typing import reveal_type

from jax.random import KeyArray

print(KeyArray)
reveal_type(KeyArray)
```
produces the output:
```
<class 'jax._src.prng.PRNGKeyArray'>
Runtime type is 'PRNGKeyArrayMeta'
```
but PyRight says:
```
/home/neil/src/tjax/a.py
  /home/neil/src/tjax/a.py:6:13 - information: Type of "KeyArray" is "Any"
0 errors, 0 warnings, 1 information 
```
Changing the value of `useLibraryCodeForTypes` doesn't seem to affect it.

I'm not sure why PyRight isn't picking up the symbol type anymore?

**Comment by erictraut** [comment]
Pyright is behaving correctly here.

Here's the relevant type definition in the `jax/random.py` source file:

```python
import typing
if typing.TYPE_CHECKING:
  PRNGKeyArray = typing.Any
  KeyArray = typing.Any
else:
  ...
```

Perhaps this was a recent change in jax.

**Comment by NeilGirdhar** [comment]
Ah, my mistake.  Sorry for the noise.
